src/util.ts

```diff
@@ -74,3 +77,68 @@ export function loadSimulariumFile(file: Blob): Promise<ISimulariumFile> {
             }
         });
 }
+
+export function calculateCachedSize(parsedAgentData: AgentData[]): number {
+    let totalSize = 0;
+
+    // Calculate the size of parsedAgentDataArray
+    for (let j = 0; j < parsedAgentData.length; ++j) {
+        const agent = parsedAgentData[j];
+        totalSize += BYTES_PER_AGENT; // 10 number properties in AgentData (excluding subpoints)
```

Comment:
this comment makes me think that you should have something like AgentData.getSizeBytes() which adds the subpoints internally.

Reply:
also there are more generic ways to do this where you don't have to worry about the object changing its number of properties, like looping over Object.keys and using typeof and hasOwnProperty... I am certain there's a npm library that would do it.  But your custom way is sure to be quicker and more efficient - at a cost of remembering to change stuff if you ever modify the AgentData class.

Reply:
I've got myself confused about my own code apparently, but this function doesn't seem to get called anymore, which makes sense because we aren't caching parsed AgentData anymore, sorry you gave it such thorough review I think it can be deleted...